packages/Angular/explorer-core/src/lib/list-component/list-component.ts

```diff
@@ -0,0 +1,200 @@
+  private list: Array<List> = [];
+  private selectedEntity = new List();
+  private selectedRow: List | null = null;
+  protected _contextUser: object | null;
+  public opened: boolean = false;
+  public deleteDialogOpened: boolean = false;
+  public selectedEntities: Array<Entity> = [];
+  public mySelection: string[] = [];
+  public loader: boolean = false;
+  public btnLoader: boolean = false;
+  public listName: string = ''
+  public listDescription: string | null = ''
+  public title: string = ''
+
+  constructor(private notificationService: NotificationService, private router: Router) {
+    this._contextUser = null
+   }
+
+  get getEntities() {
+    return this.entities
+  }
+
+  get getList() {
+    return this.list
+  }
+
+  private appendObj(listObj: List) {
+    const transformed = {
+      ID: listObj.ID,
+      Name: listObj.Name,
+      Description: listObj.Description,
+      Entity: listObj.Entity,
+      EntityID: listObj.EntityID,
+      User: listObj.User
+    }
+    if(this.selectedRow) {
+      const index: number = this.list.findIndex((item) => item.ID === listObj.ID)
+      Object.assign(this.list[index], transformed)
+    }
+    else this.list = [ ...this.list, transformed]
+       
+  }
+
+  private updateList(listID: number) {
+    this.list = this.list.filter((item) => item.ID !== listID)
+  }
+
+  private resetForm() {
+    this.listName = ''
+    this.listDescription = ''
+    this.selectedEntities = []
+    this.selectedRow = null
+  }
+
+  async ngOnInit(): Promise<void> {
+    await this.loadLists()
+    await this.loadEntities()
+  }
+
+  public async loadLists(): Promise<void> {
+    this.loader = true
+    const rv = new RunView();
+    const models = await rv.RunView({EntityName: 'Lists', ResultType: 'entity_object'}, this.md.CurrentUser);
+    this.list = models?.Results.map((entity) => {
+    return {
+      ID: entity.ID,
+      Name: entity.Name,
+      Description: entity.Description,
+      Entity: entity.Entity,
+      EntityID: entity.EntityID,
+      User: entity.User,
+      EntityInfo: entity.EntityInfo
+    }
+    });
+    this.loader = false;
+  }
+
+  public async loadEntities(): Promise<void> {
+    const rv = new RunView();
+    const models = await rv.RunView({EntityName: 'Entities', ResultType: 'entity_object'}, this.md.CurrentUser);
+    this.entities = models?.Results.map((entity) => {
+    return {
+      id: entity.ID,
+      name: entity.Name
+    }
+  });
+  }
+
+  public async saveList(): Promise<void> {
+    this.btnLoader = true
+    const md = new Metadata();
+    const listPromise = md.GetEntityObject<ListEntity>('Lists');
+    const list = await listPromise;
+    if(this.selectedRow?.ID) 
+      await list.Load(this.selectedRow.ID)
+    else 
+      list.NewRecord();
+    list.Name = this.listName;
+    list.Description = this.listDescription;
+    list.UserID = md.CurrentUser.ID;
+    list.EntityID = this.selectedEntities[0].id;
+
+    try {
+        await list.Save();
```

Comment:
The save function returns a boolean to indicate if the save was successful or not. You should check that before proceeding